internal/bminventory/inventory.go

```diff
@@ -5202,7 +5202,7 @@ func (b *bareMetalInventory) V2DownloadInfraEnvFiles(ctx context.Context, params
 	var content, filename string
 	switch params.FileName {
 	case "discovery.ign":
-		content, err = b.IgnitionBuilder.FormatDiscoveryIgnitionFile(ctx, infraEnv, b.IgnitionConfig, false, b.authHandler.AuthType(), ignition.DiscoveryISOTypeUseInfraEnv)
+		content, err = b.IgnitionBuilder.FormatDiscoveryIgnitionFile(ctx, infraEnv, b.IgnitionConfig, false, b.authHandler.AuthType(), ignition.DiscoveryISOType(*params.DiscoveryIsoType))
```

Comment:
If this can be nil we should be checking for that.
I don't think the boilerplate generated code actually fills in defaults for us.